### DISCUSSION TITLE
Sharing environment variables across multiple projects in monorepos

### DISCUSSION THREAD

**Discussion by tomdohnal** [vercel/vercel#5469] [Help] [answered]
Hello,

I have a monorepo with 3 separate apps which I want to deploy to Vercel. As per the docs I need to create 3 separate projects in Vercel for that. 

Nevertheless, I some common env vars in all three projects. Is there a way to set an environment variable for all three projects at one place? Or do I need to set it separately inside each individual project?

Thanks!

**Comment by mcsdevv** [comment, marked as answer]
Hello @tomdohnal, you'll need to set each project up separately, however, each project can reference the same secret value: https://vercel.com/docs/platform/environment-variables#secret-environment-variables

**Comment by tomdohnal** [comment]
Thanks for the answer.

Nonetheless, I still believe it'd be useful to be able to set env vars across multiple projects at one place. Should I create a separate discussion as a feature request?

**Comment by tcolinpa** [comment]
Dealing with this right now, it'd be great to have .env variables shared with all apps.

**Comment by leo** [comment]
Thank you for sharing this! Quick question: Did the three apps originate from the same source code and were later separated?

**Reply by tcolinpa** [reply]
In my case, I had two different repos running Nextjs (already separated).

My problem was inconsistency in theming and a lot of repeated code. My solution was to create a monorepo with yarn workspaces: the 2 original ones and a new one that shares theming and components across the other 2. It works great, however, it just misses this shared .env in the root.

**Comment by bobbyflowstate** [comment]
Any update on this? I'm currently switching to a mono repo and creating env variables for each Next.js project is not ideal.

**Reply by amyegan** [reply]
The team is currently working on a feature to address this and I expect it to go live _very_ soon! Keep an eye on the [changelog](https://vercel.com/changelog) 😉

**Reply by amyegan** [reply]
🎉  [Shared Environment Variables](https://vercel.com/changelog/share-environment-variables-across-your-team-and-projects) are now available to all Teams on the Pro and Enterprise plans!

**Reply by tomevans18** [reply]
This is a major win but it feels like multiple environments wasn't considered.
See my comment here https://github.com/orgs/vercel/discussions/122#discussioncomment-4119263

**Reply by amyegan** [reply]
This is good feedback Tom! I'm sharing it with the team and I'll let you know when I have more info

**Reply by amyegan** [reply]
There is not currently a way to set branch-specific shared environment variables due to complexity. 😥 

The good news is that the team has an internal ticket to make branch-specific variables work in a future iteration. But they will need to take time to do some careful planning for a stable long-term solution.
